ts/Data/DataPool.ts

```diff
@@ -299,16 +299,20 @@ class DataPool implements DataEvent.Emitter {
                 throw new Error(`Connector type not found. (${options.type})`);
             }
 
-            const connector = new ConnectorClass(
+            const connector = this.connectors[options.id] = new ConnectorClass(
                 options.options,
                 options.dataTables
             );
 
             // eslint-disable-next-line @typescript-eslint/no-floating-promises
             connector
                 .load()
-                .then((connector): void => {
-                    this.connectors[options.id] = connector;
+                .then(({ converter, dataTables }): void => {
+                    connector.dataTables = dataTables;
+                    if (converter) {
```

Comment:
It's always true, because you use `connector.dataTables` above. In case of undefined, the error will be earlier.